### ISSUE TITLE
Accuracy Mismatch and Getting Non-Integer for val_true_positives, val_true_negatives, val_false_positives, val_false_negatives

### ISSUE THREAD

**Issue by busFred** [tensorflow/tensorflow#47963]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: no
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.4.1
- Python version: 3.8.5
- CUDA/cuDNN version: 11.0
- GPU model and memory: It's on school cluster, I honestly don't know

**Describe the current behavior**
```
Epoch 99/100
43/43 [==============================] - 1s 33ms/step - loss: 0.0065 - accuracy: 0.7989 - true_positives: 20843.3636 - true_negatives: 2135.0682 - false_positives: 2.7955 - false_negatives: 0.4091 - val_loss: 0.0309 - val_accuracy: 0.7996 - val_true_positives: 9797.0000 - val_true_negatives: 965.0000 - val_false_positives: 41.0000 - val_false_negatives: 13.0000
Epoch 100/100
43/43 [==============================] - 1s 33ms/step - loss: 0.0064 - accuracy: 0.8016 - true_positives: 20843.3636 - true_negatives: 2135.0682 - false_positives: 2.7955 - false_negatives: 0.4091 - val_loss: 0.0301 - val_accuracy: 0.7955 - val_true_positives: 9796.0000 - val_true_negatives: 965.0000 - val_false_positives: 41.0000 - val_false_negatives: 14.0000
```

**Describe the expected behavior**
Metrics val_true_positives, val_true_negatives, val_false_positives, and val_false_negatives should be integers bigger than or equal to zero, but I am getting non-integer.
In addition, "accuracy =/= (true_positives+true_negatives)/(true_positives+true_negatives+false_positives+false_negatives)".

**Standalone code to reproduce the issue**
```
model: tf.keras.models.Model = tf.keras.Sequential([
    tf.keras.layers.Conv2D(
        filters=32,
        kernel_size=(20, 20),
        activation='relu',
        padding='valid',
        input_shape=train_dataset.element_spec[0].shape[1:]
    ),
    tf.keras.layers.MaxPool2D(
        pool_size=(2, 2)
    ),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(
        units=32,
        activation='relu'
    ),
    tf.keras.layers.Dense(
        units=1,
        activation='sigmoid'
    )
])
model.compile(
    optimizer='adam',
    loss='binary_crossentropy',
    metrics=[
        tf.keras.metrics.Accuracy(),
        tf.keras.metrics.TruePositives(),
        tf.keras.metrics.TrueNegatives(),
        tf.keras.metrics.FalsePositives(),
        tf.keras.metrics.FalseNegatives()
    ],
)
```

**Other info / logs**

**Comment by busFred** [comment]
After doing more Google search, I realize the problem is that I didn't really understand what precision and accuracy really means.
